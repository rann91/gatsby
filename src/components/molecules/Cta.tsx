/* @jsx jsx */
import { Box, Container, Heading, jsx, Text } from 'theme-ui'
import { Strapi_ComponentContentCta } from '../../typings/graphql'
import Link from '../atoms/Link'
import Section from '../atoms/Section'
import TextContainer from '../atoms/TextContainer'

const Cta = ({ title, description, link }: Strapi_ComponentContentCta) => (
  <Section hasTitle={!!title} sx={{ py: [7, null, null, 8], bg: 'muted' }}>
    <Container>
      <Box px={5}>
        <TextContainer>
          {title && <Heading variant="h2">{title}</Heading>}
          {description && (
            <Text
              as="p"
              variant="largeText"
              sx={{
                '* + &': {
                  paddingTop: [5]
                }
              }}>
              {description}
            </Text>
          )}
          {link && (
            <Link
              link={link}
              sx={{
                variant: 'buttons.primaryLarge',
                '* + &': {
                  mt: [6]
                }
              }}
            />
          )}
        </TextContainer>
      </Box>
    </Container>
  </Section>
)

export default Cta
