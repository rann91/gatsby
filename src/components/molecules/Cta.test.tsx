import { render } from '@testing-library/react'
import React from 'react'
import Cta from './Cta'
import data from '../../json/cta.json'

describe('Cta', () => {
  it('renders correctly', () => {
    const { container } = render(<Cta {...data} />)

    expect(container).toMatchSnapshot()
  })
})
