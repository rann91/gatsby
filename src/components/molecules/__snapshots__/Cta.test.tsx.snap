// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cta renders correctly 1`] = `
<div>
  <section
    class="css-fvc22l-Cta"
    data-testid="section"
  >
    <div
      class="css-1yrmf69"
    >
      <div
        class="css-i1z8pl"
      >
        <div
          class="css-ycxng9-TextContainer"
          data-testid="text-container"
        >
          <h2
            class="css-1wf7gym"
          >
            Grow your business
          </h2>
          <p
            class="css-w9nyfe-Cta"
          >
            Today is the day to build the business of your dreams. Share your mission with the world â€” and blow your customers away.
          </p>
          <a
            class="css-1s79t5o-Cta"
            data-testid="gatsby-link"
            href="/about"
            target="_self"
          >
            Learn more
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
